File: frontend/src/types/report.ts
export type ReportPeriod = 'daily' | 'weekly' | 'monthly'
export type ReportExportFormat = 'pdf' | 'excel'

export interface ReportDayEntry {
  date: string
  tasks_completed: number
  habits_completed: number
}

export interface ReportSummary {
  period: ReportPeriod
  start_date: string
  end_date: string
  tasks_completed: number
  tasks_due: number
  task_completion_rate: number
  habits_active: number
  habit_checkins: number
  habit_completion_rate: number
  goals_completed: number
  best_streak: number
  daily_breakdown: ReportDayEntry[]
}

File: backend/app/schemas/report.py
from datetime import date
from enum import Enum

from pydantic import BaseModel


class ReportPeriod(str, Enum):
    daily = "daily"
    weekly = "weekly"
    monthly = "monthly"


class ReportExportFormat(str, Enum):
    pdf = "pdf"
    excel = "excel"


class ReportDayEntry(BaseModel):
    date: date
    tasks_completed: int
    habits_completed: int


class ReportSummary(BaseModel):
    period: ReportPeriod
    start_date: date
    end_date: date
    tasks_completed: int
    tasks_due: int
    task_completion_rate: float
    habits_active: int
    habit_checkins: int
    habit_completion_rate: float
    goals_completed: int
    best_streak: int
    daily_breakdown: list[ReportDayEntry]
